src/app/CommandHandler.cpp

```diff
@@ -125,6 +119,14 @@ void CommandHandler::Close()
     mSuppressResponse = false;
     MoveToState(CommandState::AwaitingDestruction);
 
+    // We must finish all async work before we can shut down a CommandHandler. The actual CommandHandler MUST finish their work
+    // in reasonable time or there is a bug.
+    //
+    // The only case for releasing CommandHandler without CommandHandler::Handle releasing its reference is the stack shutting down,
+    // in which case Close() is not called.
+    VerifyOrDieWithMsg(mRefCount == 0, DataManagement, "CommandHandler::Close() called with %zu unfinished async work items",
```

Comment:
Shouldn't this be removed now?

Reply:
Updated the comment, instead of bug, calling Close() with `mRrfCount != 0` should be something never happen. I want to keep this check as a assumption, just like the similar checks in PacketBufferHandle